### PR TITLE
CMake based build for MSVC 13 and 15

### PR DESCRIPTION
**Pull request by yuslepukhin**
Note: this does not fixes bugs but builds tests
  so we can see where bugs are.
  None of them are fixes in this iteration. I deemed OK to suppress
  warnings that I have seen.
  The benchmark is not currently built due to linking issues
  and bugs are currently a priority.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,697 @@
...
+set(JEMALLOC_HAVE_SECURE_GETENV 0)
+
+set(JEMALLOC_HAVE_ISSETUGID 0)
+set(JEMALLOC_MALLOC_THREAD_CLEANUP 0)
+
+set(JEMALLOC_MUTEX_INIT_CB 0)
+
+# No lazy lock since there is a way in Windows
+# to bootstrap for sure
+set(JEMALLOC_LAZY_LOCK 0)
```
Comment (yuslepukhin) [inline CMakeLists.txt]: Now, I am missing je_init in this version which is currently invoked in our in-house version before any static data is initialized.

**Comment by yuslepukhin** [comment]
@rustyx I hope this would help you iterate.

**Comment by rustyx** [comment]
Wouldn't it be confusing to have both CMake and Autotools config in one project?
IMO if you want to introduce CMake, then it should be done for all targets, not just for Windows. Otherwise the risk is high that it'll become a maintenance nightmare for us Windows users.

**Comment by yuslepukhin** [comment]
@rustyx  For windows users installing Cygwin and related things might be a little more effort than simply CMake. The dual approach works allright for RocksDB but it is your call. I will keep this open until tomorrow.

**Pull request closed without merging**